src/controls/controller.h

```diff
@@ -1,7 +1,27 @@
 #ifndef _CONTROLLER_H
 #define _CONTROLLER_H
 
-#include <ultra64.h>
+#include <stdint.h>
+
+// The button values match those defined by libultra - see os_cont.h. Instead of
+// hard-coding these values here, a better approach would be to have a dedicated
+// file for each library implementation and choose the appropriate one at build
+// time.
+// NOTE: libdragon uses the same values but reversed - see joypad_buttons_t.
+#define BUTTON_A       0x8000
+#define BUTTON_B       0x4000
```

Comment:
Libultra values are fine for now, but what I was imagining eventually was a generic interface that didn't change with the implementation. So library-independent controller constants which are converted to on read (see my other comment regarding reuse of common code). This doesn't need to be changed now.

Reply:
Totally agree.